### PR TITLE
chore: add tooling to renew stata license

### PR DESCRIPTION
**Pull request by bloodearnest**
- copy the stinit binary from the stata into the image
 - tooling and documentation to generate a new license file.

stinit has no automation support, so we use expect to pretend to be
a human.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,10 +16,31 @@ To build, install Stata locally, apply your licence, and then copy the
...
 * `/usr/local/stata/utilities/`
 
+
 ## Relicensing Stata
 
-To relicense, apply the licence locally, and then copy
-`/usr/local/stata/stata.lic` into `bin/`.
+The docker image contains the machinery needed to update the license. To
+generate a new license file, first obtain the new serial number, code, and
+authorisation. Then run:
```
Comment (sebbacon) [inline README.md:28]: Worth including some examples of what each one should look like? (For example, I'm thinking the serial number probably has to include spaces?)

File: README.md [hunk 2]
```diff
@@ -16,10 +16,31 @@ To build, install Stata locally, apply your licence, and then copy the
...
+    docker run --rm -v $PWD:/src -w /usr/local/stata --entrypoint /src/scripts/renew-license.sh ghcr.io/opensafely-core/stata-mp "SERIAL" "CODE" "AUTH"
+
+The resulting file will be copied to `./stata.lic` (note: due to docker
+shenanigans, it will be owned by root). You can test this works with:
+
+    ./scripts/test-license.sh
+
+The contents of this license file can then be used to update the `STATA_LICENSE`
+env var for job-runner. You should also update the `stata.lic` file in
+[https://github.com/opensafely/server-instructions](https://github.com/opensafely/server-instruction)
```
Comment (sebbacon) [inline README.md:39]: I couldn't remember what `server-instructions` is -- could you add a short README in that repo?

**Review by evansd — APPROVED** [review]
Excellent use of `expect` A+++

**Review by sebbacon — APPROVED** [review]
Lovely

File: README.md (hunk 1 above)
Comment (bloodearnest) [reply README.md]: Good, point, and excellent example, as it's `code` that has spaces!

Have added explicit examples

File: README.md (hunk 2 above)
Comment (bloodearnest) [reply README.md:45]: Hmm, this is a bit tricky.

I believe originally it served as docs for how to access TPP server, but it's badly out of date.

I stuck the license file in there as I needed a private repo that only opensafely org members could access, so they can grab our stata license locally when needed.

So I'm not 100% sure what the read me should say, other than ^

Comment (sebbacon) [reply README.md:45]: I think that's better than nothing :D

Comment (bloodearnest) [reply README.md:45]: done

**Pull request merged**